error TS2304: Cannot find name 'AsyncGenerator'

I have a:

2. [x] Issue:
  * [x] **Provide error messages including stacktrace**
  * [ ] Provide a **minimal** sample reproduction. **Create a reproduction based on this [sandbox](https://codesandbox.io/s/v3v0my2370)**
  * [x] Did you check this issue wasn't filed before?
  * [x] Elaborate on your issue. What behavior did you expect?
  * [x] State the versions of MobX and relevant libraries. Which browser / node / ... version?

After update to mobx 4.15.0 I have this issue (4.14.1 works ok):
```
node_modules/mobx/lib/api/flow.d.ts:9:108 - error TS2304: Cannot find name 'AsyncGenerator'.

9 export declare function flow<R, Args extends any[]>(generator: (...args: Args) => Generator<any, R, any> | AsyncGenerator<any, R, any>): (...args: Args) => CancellablePromise<R>;
                                                                                                             ~~~~~~~~~~~~~~


Found 1 error.
```

My environment:
node: 13.1.0
typescript: 3.7.2

tsconfig.json (related parts):
```json
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "jsx": "react-native",
    "lib": ["es2015", "es2016"],
    "module": "es6",
    "moduleResolution": "node",
    "noEmit": true,
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "noImplicitReturns": true,
    "noUnusedLocals": true,
    "preserveSymlinks": true,
    "resolveJsonModule": true,
    "strict": true,
    "target": "es6",
    "importHelpers": true,
    "types": ["react", "react-native"]
  },
}
```

Above error because AsyncGenerator defined only in lib.es2018.asyncgenerator.d.ts

So question, does mobx (even 4.x) now requires es2018 to work properly?

--- a/docs/faq/faq.md
+++ b/docs/faq/faq.md
@@ -12,6 +12,14 @@ hide_title: true
 
 MobX runs on any ES5 environment. That means that Node.js, Rhino and all browsers except for IE8 are supported. See [caniuse.com](http://caniuse.com/#feat=es5)
 
+##### While using Typescript I get `error TS2304: Cannot find name 'AsyncGenerator'`
+
+Edit your `tsconfig.json` and ensure your `lib` section array includes `es2018.asynciterable` or one of its super-sets (e.g. `es2018` or `esnext`). Note that this won't affect code generation in anyway, it just affects the standard type definitions the Typescript compiler will use, which should be a safe change. If your `tsconfig.json` does not include a lib section the defaults are:
+* ES5 target: DOM,ES5,ScriptHost
+* ES6 target: DOM,ES6,DOM.Iterable,ScriptHost
+
+So you'd need to add `es2018.asynciterable` to those defaults in this particular case.
+
 ##### Can MobX be combined with RxJS?
 
 Yes, you can use [toStream and fromStream from mobx-utils](https://github.com/mobxjs/mobx-utils#tostream) to use RXJS and other TC 39 compatible observables with mobx.

